core/src/main/java/org/bitcoinj/core/PeerSocketHandler.java

```diff
@@ -155,8 +155,6 @@ public int receiveBytes(ByteBuffer buff) {
                 Message message;
                 int preSerializePosition = buff.position();
                 try {
-                    // The peer will send us a message based on our protocol version, but theirs may be lower than ours
-                    serializer.setProtocolVersion(peerAddress.protocolVersion);
```

Comment:
This is not thread safe (each connected peer or node is managed by a separate thread, but there is only one serializer object handling all threads and nodes which may have different protocol versions).